src/main/java/hospital/userservice/service/UserServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package hospital.userservice.service;
+
+import hospital.userservice.dto.request.CreateUserRequest;
+import hospital.userservice.dto.response.UserResponse;
+import hospital.userservice.exception.NotFoundException;
+import hospital.userservice.mapper.UserMapper;
+import hospital.userservice.model.Department;
+import hospital.userservice.model.Permission;
+import hospital.userservice.model.User;
+import hospital.userservice.repository.DepartmentRepository;
+import hospital.userservice.repository.PermissionRepository;
+import hospital.userservice.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService{
+
+    private final UserRepository userRepository;
+    private final DepartmentRepository departmentRepository;
+    private final PermissionRepository permissionRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponse createUser(CreateUserRequest createUserRequest) {
+
+        System.out.println(createUserRequest.getDepartmentId());
+        Department department = departmentRepository.findById(createUserRequest.getDepartmentId())
+                .orElseThrow(() -> {
+                        throw new NotFoundException("Department with given id doesn't exist");
+                    }
+                );
+
+        List<Permission> permissions = permissionRepository.findPermissionsByNameIsIn(Arrays.asList(createUserRequest.getPermissions()));
```

Comment:
I think this findPermissionsByNameIsIn could be named better, it's not obvious what a function does. Maybe something like checkIfUserHaveThePermission? If that's what this function does.

Reply:
We cant rename that function since naming it gives that repository appropriate sql query behind the scenes.